Tetrahedral_remeshing/include/CGAL/Tetrahedral_remeshing/Remeshing_triangulation_3.h

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2020 GeometryFactory (France) and Telecom Paris (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Jane Tournois, Noura Faraj
+
+#ifndef CGAL_TETRAHEDRAL_REMESHING_TRIANGULATION_H
+#define CGAL_TETRAHEDRAL_REMESHING_TRIANGULATION_H
+
+#include <CGAL/license/Tetrahedral_remeshing.h>
+
+#include <CGAL/Triangulation_data_structure_3.h>
+#include <CGAL/Triangulation_3.h>
+
+#include <CGAL/Tetrahedral_remeshing/Remeshing_cell_base_3.h>
+#include <CGAL/Tetrahedral_remeshing/Remeshing_vertex_base_3.h>
+
+#include <CGAL/Kernel_traits.h>
+#include <CGAL/Cartesian_converter.h>
+#include <CGAL/tags.h>
+
+#include <boost/static_assert.hpp>
+#include <boost/type_traits/is_same.hpp>
+
+namespace CGAL
+{
+namespace Tetrahedral_remeshing
+{
+/*!
+\ingroup PkgTetrahedralRemeshingClasses
+
+The class `Remeshing_triangulation_3`
+is a class template which provides a valid triangulation type
+that can be used as the 3D triangulation input for
+the tetrahedral remeshing process.
+
+\tparam Gt is the geometric traits class.
+It has to be a model of the concept `RemeshingTriangulationTraits_3`.
+
+\tparam Concurrency_tag enables sequential versus parallel implementation of the
+triangulation data structure.
+Possible values are `Sequential_tag` (the default) and `Parallel_tag`.
+
+\tparam Vb is a vertex base class from which `Remeshing_vertex_base_3` derives.
```

Comment:
How come the vertex/cell parameter templates of this triangulation are the bases of the actual vertex/cell types? Maybe to clarify what I mean: the cell base of `Delaunay_triangulation_3` will have concept `DelaunayTriangulationCellBase_3`, not `TriangulationCellBase_3` while wrapping that cell in a `Delaunay_cell_base_3` within the triangulation.

It's also a bit confusing since the main function asks the `TDS` to be model of `TriangulationDataStructure_3`, with cell base model of `MeshCellBase_3` and vertex base model of `MeshVertexBase_3`, so you have to figure out that you need to use that triangulation, which is kinda awkward.

Reply:
I understand your point, but it is exactly what it is...
`Remeshing_cell_base_3` derives from `Mesh_cell_base_3` (actually it is exactly it), so, as `Mesh_cell_base_3`, it derives from its `Vb` template parameter.
Maybe we should discuss that phrasing further though.

Reply:
This is the template parameter of the triangulation here, not the remeshing vertex base class. In comparison, `Mesh_triangulation_3` has
```
Vertex_base | must be a model of MeshVertexBase_3 or Default and defaults to Mesh_vertex_base_3<Gt, MD>.
```
not 
```
Vertex_base | is a vertex base class from which Mesh_vertex_base_3<Gt, MD, Vertex_base> derives and must be a model of TriangulationVertexBase_3 or Default and defaults to Triangulation_vertex_base_3<Gt, MD>.
```
which would be the equivalent of how things are introduced here currently.